CONVENTIONS.md

```diff
@@ -15,6 +16,12 @@
   - [Identifiers](#identifiers)
   - [Variables](#variables)
 
+## Plugin Location
+
+- Most plugins should start in a repository outside of the OpenSearch project. This allows for rapid iteration during initial development without the overhead of being an official part of the project.
+- Plugins can be later ported into the OpenSearch Project on a case-by-case basis. A formal path for this process is yet to be created.
+- Do not add new plugins to the [plugins folder](https://github.com/opensearch-project/OpenSearch/tree/main/plugins) within the core OpenSearch repository. The maintainers likely do not have domain expertise in your new optional plugin. If you think it should be there anyway, please explain why in the RFC stage. In any case, it is recommended to start outside the project.   
```

Comment:
"In any case, it is recommended to start outside the project. " is kinda redundant.